queries.sql

```diff
@@ -7,4 +7,70 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/*--- Transactions ---*/
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+
+SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals 
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+SELECT * FROM animals;
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT my_savepoint;
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 1;
+COMMIT;
+SELECT * FROM animals;
+
+/*--- Queries ---*/
+
+SELECT COUNT(*) 
+FROM animals;
+
+SELECT COUNT(*) 
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg)
+FROM animals;
+
+SELECT neutered, SUM(escape_attempts) 
+FROM animals 
+GROUP BY neutered;
+
+SELECT species, 
+MIN(weight_kg), MAX(weight_kg)
+FROM animals
+GROUP BY species;
+
+SELECT species,
+AVG(escape_attempts)
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '1999-12-31'
```

Comment:
- [ ] In creating a query for the question: _**What is the average number of escape attempts per animal type of those born between 1990 and 2000?**_ our query should look like:
```sql
   SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
```

Reply:
Hi @rloterh,  thank you for your constructive comments.